test/integration_test.exs

```diff
@@ -0,0 +1,138 @@
+defmodule Membrane.RTC.Engine.TimescaleDB.IntegrationTest do
+  use ExUnit.Case
+
+  alias Membrane.RTC.Engine.TimescaleDB.{
+    Cleaner,
+    Model,
+    Repo
+  }
+
+  alias Model.{
+    PeerMetrics,
+    TrackMetrics
+  }
+
+  setup do
+    Ecto.Adapters.SQL.Sandbox.checkout(Repo)
+    :ok
+  end
+
+  describe "Reporter stores" do
+    test "report in the db" do
+      Model.insert_report(Repo, rtc_engine_report_1())
+
+      assert Repo.all(PeerMetrics) |> length() == 2
+      assert Repo.all(TrackMetrics) |> length() == 3
+
+      Ecto.Adapters.SQL.Sandbox.checkin(Repo)
+    end
+
+    test "multiple reports in the db" do
+      Model.insert_report(Repo, rtc_engine_report_1())
+      Model.insert_report(Repo, rtc_engine_report_2())
+
+      assert Repo.all(PeerMetrics) |> length() == 3
+      assert Repo.all(TrackMetrics) |> length() == 4
+
+      Ecto.Adapters.SQL.Sandbox.checkin(Repo)
+    end
+  end
+
+  describe "Cleaner removes" do
+    @tag :long_running
+    test "old vales from the db" do
+      {:ok, cleaner} =
+        Cleaner.start_link(cleanup_interval: 1, metrics_lifetime: 4, repo: Repo, name: Cleaner)
+
+      Ecto.Adapters.SQL.Sandbox.allow(Repo, self(), cleaner)
+
+      Model.insert_report(Repo, rtc_engine_report_1())
+
+      Process.sleep(3000)
```

Comment:
These sleeps are quite long. Maybe you could manipulate the Cleaner settings to make them shorter

Reply:
Currently Cleaner expects `metrics_lifetime` and `cleanup_interval` expressed in seconds, so without changing that, it is hard to manipulate these values, without making test flaky